eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockChain.hs

```diff
@@ -242,6 +226,21 @@ txSeqDecoder lax = do
             Seq.zipWith3 segWitAnnTx bodies wits metadata
   pure $ TxSeq' <$> txns <*> bodiesAnn <*> witsAnn <*> metadataAnn
 
+auxDataSeqDecoder ::
+  Int -> IntMap (m (TxAuxData era)) -> Bool -> Decoder s (Seq (Maybe (m (TxAuxData era))))
```

Comment:
This will allow you to avoid using `runIdentity`
```suggestion
  Int -> IntMap a -> Bool -> Decoder s (Seq (Maybe a))
```

Reply:
Nice, didn't realize that TxAxuData type is not relevant to the function at all. Thanks!